.github/workflows/test.yml

```diff
@@ -30,6 +30,35 @@ jobs:
         run: pnpm test:ember
         working-directory: test-app
 
+  typescript-compatibility:
+    name: Type checking - ${{ matrix.typescript-scenario }}
+    runs-on: ubuntu-latest
+
+    needs: [test]
+
+    strategy:
+      fail-fast: false
+      matrix:
+        typescript-scenario:
+          - typescript@5.0
+          - typescript@5.1
+          - typescript@next
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install pnpm
+        uses: wyvox/action-setup-pnpm@v3
+        with:
+          node-version: 16.x
+          args: "--frozen-lockfile"
+      - name: Update TS version
+        run: pnpm add -D ${{ matrix.typescript-scenario }}
+        working-directory: ember-phone-input
+      - name: Type checking
+        run: pnpm lint:types
+        working-directory: ember-phone-input
```

Comment:
Maybe we should move these jobs in the `lint.yml` script? Since it's not about tests but types linting

Reply:
Yes actually that's what I did at first but not sure why I moved them to test. Will update anyway!